sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala

```diff
@@ -965,7 +966,8 @@ class MicroBatchExecution(
       updateStateStoreCkptId(execCtx, latestExecPlan)
     }
     execCtx.reportTimeTaken("commitOffsets") {
-      if (!commitLog.add(execCtx.batchId, CommitMetadata(watermarkTracker.currentWatermark))) {
+      if (!commitLog.add(execCtx.batchId,
+        CommitMetadata(watermarkTracker.currentWatermark, currentStateStoreCkptId.toMap))) {
```

Comment:
Same as https://github.com/apache/spark/pull/48355#discussion_r1865926695

Reply:
> Did you validate that for V1, the commit log generated can be read by an older version?
Yes this is verified in the previous PR

Reply:
@WweiL the added the test is great. This is not what I meant in the comment though. I didn't mean downgrading to V1 in the same release, but downgrade to an older release where the code of `CommitMetadata` doesn't have the extra column added here.

Reply:
Further clarify: I mean the commit log generated by the new code with V1 should be readable by older code. It is likely to be OK either way (if we doesn't work we need to bump up the version), but I think having a Noneable field is more straight-forward.

Reply:
Never mind. I was wrong. I missed that you created `class CommitLogLegacy` in your unit test, so we are all good!

Reply:
@WweiL there must be a misunderstanding here. My previous comment
> Same as commented above. It is better to fill a None if it is it is V1.
> Did you validate that for V1, the commit log generated can be read by an older version?

is two different comments. Even if validation for V1 is done, I still hope None is used for V1, and I've explained it above. See my explanation above https://github.com/apache/spark/pull/48355/files#r1870082653
I didn't see it addressed yet.

Reply:
@siying I will address this in another PR, depending on if this is merged or yet I might create a separate one:
https://github.com/apache/spark/pull/49063
Let's scope this PR and ignore this for now

Reply:
@WweiL I perhaps overlooked it in the previous code review, but this is a format change, so we need to fix it as soon as we can. We can separate this to another PR if it can make the format fix merged sooner.